### PR TITLE
Update multicall

### PR DESCRIPTION
**Pull request by polarker**
To have better types for multicall as requested by reed

### CODE REVIEW

**Comment by polarker** [comment]
This does not work as expected due to limits of TS compiler

**Comment by polarker** [comment]
@h0ngcha0 This is now ready for review!

**Review by h0ngcha0 — APPROVED** [review]
LGTM :+1:  only one comment

File: packages/web3/src/utils/utils.ts [hunk 1]
```diff
@@ -169,3 +169,11 @@ export type Eq<X, Y> = _Eq<{ [P in keyof X]: X[P] }, { [P in keyof Y]: Y[P] }>
 // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
 export function assertType<T extends true>(): void {}
 export type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>
+
+export type Narrow<type> =
+  | (unknown extends type ? unknown : never)
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | (type extends Function ? type : never)
+  | (type extends bigint | boolean | number | string ? type : never)
+  | (type extends [] ? [] : never)
```
[the remark this answers is not in the document]
Comment (polarker) [reply packages/web3/src/utils/utils.ts:178]: He reverted that comment in another post. I will leave it untouched. It's the same code used in `Viem`

**Pull request merged**